### PR TITLE
add event map to AppHistory, for types of Event

### PR DESCRIPTION
**Pull request by samthor**
This will give types when writing code like:

```js
window.appHistory.addEventListener('navigate', (event) => {
  // "event" is known by TS to be of type AppHistoryNavigateEvent
});
```

### CODE REVIEW

File: app_history.d.ts [hunk 1]
```diff
@@ -2,6 +2,11 @@ interface Window {
   readonly appHistory: AppHistory;
 }
 
+interface AppHistoryEventMap {
+  "navigate": AppHistoryNavigateEvent;
+  "navigateerror": ErrorEvent;
```
Comment (domenic) [inline app_history.d.ts:7]: Should we also add `navigatesuccess`? That's just `Event`, but it looks like that's still reasonably common: e.g. https://github.com/microsoft/TypeScript/blob/9cc30709380da8b212487f63d134b7be9b001511/lib/lib.dom.d.ts#L1916-L1931

**Review by domenic — COMMENTED** [review]
Should we do the same thing for AppHistoryEntry? The events aren't implemented (or specced) yet but it seems like we're tracking the explainer more than the implementation or spec.

**Comment by samthor** [comment]
Yeah, whether to include `Event` was a bit unclear but I see the precedent and it makes it clearer what to update if the `Event` types change. I'll update the PR.

File: app_history.d.ts (hunk 1 above)
Comment (samthor) [reply app_history.d.ts:7]: Done

**Review by domenic — APPROVED** [review]

**Pull request merged**